rust/ql/test/query-tests/security/CWE-089/sqlx.rs

```diff
@@ -61,14 +61,14 @@ async fn test_sqlx_mysql(url: &str, enable_remote: bool) -> Result<(), sqlx::Err
     let prepared_query_1 = String::from("SELECT * FROM people WHERE firstname=?"); // (prepared arguments are safe)
 
     // direct execution
-    let _ = conn.execute(safe_query_1.as_str()).await?; // $ sql-sink
-    let _ = conn.execute(safe_query_2.as_str()).await?; // $ sql-sink
-    let _ = conn.execute(safe_query_3.as_str()).await?; // $ sql-sink
-    let _ = conn.execute(unsafe_query_1.as_str()).await?; // $ sql-sink Alert[rust/sql-injection]=args1
+    let _ = conn.execute(safe_query_1.as_str()).await?; // $ MISSING: sql-sink
```

Comment:
While we are able to infer that `conn` has type `PoolConnection`, we cannot currently infer that is in fact has type `PoolConnection<MySql>` (https://github.com/github/codeql/pull/19954 or similar is needed for that).

Once we are able to do that, we need implicit dereferencing via
```
impl<DB: Database> Deref for PoolConnection<DB> {
    type Target = DB::Connection;

    fn deref(&self) -> &Self::Target {
        &self.live.as_ref().expect(EXPECT_MSG).raw
    }
}
```
to get to the type `&MySqlConnection`, and then via `impl<'c> Executor<'c> for &'c mut MySqlConnection` we can find `execute`.